File: gencode/python/udmi/schema/events.py
"""Generated class for events.json"""
from .events_system import SystemEvents
from .events_pointset import PointsetEvents
from .events_discovery import DiscoveryEvents


class Events:
  """Generated schema class"""

  def __init__(self):
    self.system = None
    self.pointset = None
    self.discovery = None

  @staticmethod
  def from_dict(source):
    if not source:
      return None
    result = Events()
    result.system = SystemEvents.from_dict(source.get('system'))
    result.pointset = PointsetEvents.from_dict(source.get('pointset'))
    result.discovery = DiscoveryEvents.from_dict(source.get('discovery'))
    return result

  @staticmethod
  def map_from(source):
    if not source:
      return None
    result = {}
    for key in source:
      result[key] = Events.from_dict(source[key])
    return result

  @staticmethod
  def expand_dict(input):
    result = {}
    for property in input:
      result[property] = input[property].to_dict() if input[property] else {}
    return result

  def to_dict(self):
    result = {}
    if self.system:
      result['system'] = self.system.to_dict() # 4
    if self.pointset:
      result['pointset'] = self.pointset.to_dict() # 4
    if self.discovery:
      result['discovery'] = self.discovery.to_dict() # 4
    return result

File: gencode/java/udmi/schema/Events.java

package udmi.schema;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;


/**
 * Events
 * <p>
 * Container object for all event schemas, not directly used.
 * 
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Events {

    /**
     * System Events
     * <p>
     * Used for system events such as logging. [System Event Documentation](../docs/messages/system.md#event)
     * 
     */
    @JsonProperty("system")
    @JsonPropertyDescription("Used for system events such as logging. [System Event Documentation](../docs/messages/system.md#event)")
    public SystemEvents system;
    /**
     * Pointset Events
     * <p>
     * A set of points reporting telemetry data. [Pointset Events Documentation](../docs/messages/pointset.md#telemetry)
     * 
     */
    @JsonProperty("pointset")
    @JsonPropertyDescription("A set of points reporting telemetry data. [Pointset Events Documentation](../docs/messages/pointset.md#telemetry)")
    public PointsetEvents pointset;
    /**
     * Discovery Events
     * <p>
     * [Discovery result](../docs/specs/discovery.md) with implicit discovery
     * 
     */
    @JsonProperty("discovery")
    @JsonPropertyDescription("[Discovery result](../docs/specs/discovery.md) with implicit discovery")
    public DiscoveryEvents discovery;
    /**
     * Streams Events
     * <p>
     * Sequential data stream chunks for reliable transport over MQTT (e.g. PCAP traces, firmware blobs, reliable alarms, logs playback).
     * 
     */
    @JsonProperty("streams")
    @JsonPropertyDescription("Sequential data stream chunks for reliable transport over MQTT (e.g. PCAP traces, firmware blobs, reliable alarms, logs playback).")
    public StreamsEvents streams;

    @Override
    public int hashCode() {
        int result = 1;
        result = ((result* 31)+((this.pointset == null)? 0 :this.pointset.hashCode()));
        result = ((result* 31)+((this.system == null)? 0 :this.system.hashCode()));
        result = ((result* 31)+((this.discovery == null)? 0 :this.discovery.hashCode()));
        result = ((result* 31)+((this.streams == null)? 0 :this.streams.hashCode()));
        return result;
    }

    @Override
    public boolean equals(Object other) {
        if (other == this) {
            return true;
        }
        if ((other instanceof Events) == false) {
            return false;
        }
        Events rhs = ((Events) other);
        return (((((this.pointset == rhs.pointset)||((this.pointset!= null)&&this.pointset.equals(rhs.pointset)))&&((this.system == rhs.system)||((this.system!= null)&&this.system.equals(rhs.system))))&&((this.discovery == rhs.discovery)||((this.discovery!= null)&&this.discovery.equals(rhs.discovery))))&&((this.streams == rhs.streams)||((this.streams!= null)&&this.streams.equals(rhs.streams))));
    }

}
